workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourManagerInitializationSystem.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Creates and removes SpatialOSBehaviourManager object for EntityGameObjects.
+    /// </summary>
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.GameObjectInitializationGroup))]
+    public class SpatialOSBehaviourManagerInitializationSystem : ComponentSystem
+    {
+        public struct AddedEntitiesData
+        {
+            public readonly int Length;
+            public EntityArray Entities;
+            public ComponentArray<GameObjectReference> GameObjectReferences;
+            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
+            [ReadOnly] public ComponentDataArray<RequiresSpatialOSBehaviourManager> RequiresSpatialOSBehaviourManagerTags;
+        }
+
+        public struct RemovedEntitiesData
+        {
+            public readonly int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
+            public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
+        }
+
+        [Inject] private AddedEntitiesData addedEntitiesData;
+        [Inject] private RemovedEntitiesData removedEntitiesData;
+
+        private GameObjectDispatcherSystem gameObjectDispatcherSystem;
+        private ILogDispatcher logDispatcher;
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+
+            gameObjectDispatcherSystem = World.GetExistingManager<GameObjectDispatcherSystem>();
```

Comment:
This has a dependency on the order in which these systems are defined in the world setup - is that intentional?

Reply:
Very interesting. You are right. The reason I'm using `GetExistingManager` here is because I got trolled before when I ran a worker instance where I intentionally didn't want the `GameObjectDispatcherSystem` running. Reverting to `GetOrCreateManager()`.